### ISSUE TITLE
[Bug]: herd composer uses wrong php version

### ISSUE THREAD

**Issue by N-Silbernagel** [beyondcode/herd-community#199]
### Operating system version

macOs Ventura 13.5

### System architecture

ARM64 (M1, M2, etc)

### Herd Version

1.2.2 (Build 16)

### Bug description

I've got an isolated site which uses PHP8.1, my default is PHP 8.2.
When doing "herd which-php" in that sites directory, I get "/Library/Application Support/Herd/bin/php81" which is correct.
When doing "herd composer ..." tho, it seems to use PHP 8.2 because it logs deprecation the notices " Callables of the form [...] are deprecated".

In my case this occurs when running a composer script to execute migrations via phinx.

These warning were not present using valet. https://github.com/laravel/valet/blob/master/valet#L168

**Comment by jacobgraf** [comment]
I wonder if this is also related to the recent issue I am seeing on my Craft CMS sites? Now, when I try to run updates from the Craft CMS web interface, I get the following error. I can run the updates fine from the CLI using composer, but it's almost like Composer can't find php to run them within the GUI?

It used to work with Herd, but recently stopped working.

```
Error: The command "'php' '/Users/jacobgraf/Sites/ghostbed-cms/storage/runtime/composer.phar' 'update' 'craftcms/cms' '--with-all-dependencies' '--working-dir' '/Users/jacobgraf/Sites/ghostbed-cms' '--no-scripts' '--no-ansi' '--no-interaction'" failed.

Exit Code: 127(Command not found)

Working directory: /Users/jacobgraf/Sites/ghostbed-cms/web

Output:
================

Error Output:
================
sh: line 0: exec: php: not found

Composer output: sh: line 0: exec: php: not found
```

**Comment by jannisborgers** [comment]
@jacobgraf I’m getting the same error after switching from Valet to Herd and trying to install a plugin from the control panel in Craft. Seems like Craft executing `php …` cannot find Herd’s PHP binary.

**Comment by samhibberd** [comment]
Has anyone worked this out? We are running into the same issue here :(

**Comment by jacobgraf** [comment]
> Has anyone worked this out? We are running into the same issue here :(

Nope. We’ve had to just use the command line for all of our Craft updates. Definitely kind of annoying.

**Comment by jacobgraf** [comment]
> Has anyone worked this out? We are running into the same issue here :(

Nope. We’ve had to just use the command line for all of our Craft updates. Definitely kind of annoying.

**Comment by samhibberd** [comment]
Same :( Will post here if we work anything out.

**Comment by N-Silbernagel** [comment]
No, I am running them in terminal

**Comment by jacobgraf** [comment]
Why did this get closed. Is there going to be a first-party fix put in place so we don't need to manually tweak the FPM configs?

**Comment by nlvedwin** [comment]
any update on this?

**Comment by nicekiwi** [comment]
it seems you have to select the global php version manually, it won't use the version in the local project, which is a shame. :(

**Comment by N-Silbernagel** [comment]
@nicekiwi im not using herd anymore, but have you tried prefixing your composer scripts with "@php"? Thats a composer feature. Say you had "phphstan" for running phpstan, try "@php phpstan".
